x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rules.test.tsx

```diff
@@ -4,24 +4,42 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
-import { renderHook, act } from '@testing-library/react-hooks';
-import { useLoadRules } from './use_load_rules';
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks/dom';
+import { useLoadRulesQuery as useLoadRules } from './use_load_rules_query';
 import {
   RuleExecutionStatusErrorReasons,
   RuleExecutionStatusWarningReasons,
 } from '@kbn/alerting-plugin/common';
 import { RuleStatus } from '../../types';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useKibana } from '../../common/lib/kibana';
+import { IToasts } from '@kbn/core-notifications-browser';
+import { waitFor } from '@testing-library/dom';
 
+jest.mock('../../common/lib/kibana');
 jest.mock('../lib/rule_api/rules_kuery_filter', () => ({
   loadRulesWithKueryFilter: jest.fn(),
 }));
 
+const useKibanaMock = useKibana as jest.Mocked<typeof useKibana>;
+
 const { loadRulesWithKueryFilter } = jest.requireMock('../lib/rule_api/rules_kuery_filter');
 
-const onError = jest.fn();
 const onPage = jest.fn();
 
+const queryClient = new QueryClient({
```

Comment:
should we create a mock client? just to cut down on the repeated code

Reply:
I don't see the need yet but it's something that we might end up needing